common/flatpak-utils.c

```diff
@@ -3860,15 +3860,22 @@ extract_appstream (OstreeRepo   *repo,
           component_id_text_node = flatpak_xml_find (component_id, NULL, NULL);
 
           component_id_text = g_strstrip (g_strdup (component_id_text_node->text));
-          if (!g_str_has_prefix (component_id_text, id) ||
-              !g_str_has_suffix (component_id_text, ".desktop"))
+
+          /* .desktop suffix in component ID is suggested, not required
+             (unless app ID actually ends in .desktop) */
+          if (g_str_has_suffix (component_id_text, ".desktop") &&
+              !g_str_has_suffix (id, ".desktop"))
+            component_id_text[strlen (component_id_text) - strlen (".desktop")] = 0;
+
+          if (!g_str_has_prefix (component_id_text, id))
             {
+              g_debug ("Skipping mismatched appstream component %s for app %s\n",
```

Comment:
`g_debug()` appends a `\n` for you, so you should not put one in the message. (It’s only `g_print()` and `g_printerr()` which don’t automatically append `\n`.)

Reply:
I thought g_print but typed g_debug due to speculative execution in my fingertips.